botorch_community/acquisition/latent_information_gain.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Latent Information Gain Acquisition Function for Neural Process Models.
+
+References:
+
+.. [Wu2023arxiv]
+   Wu, D., Niu, R., Chinazzi, M., Vespignani, A., Ma, Y.-A., & Yu, R. (2023).
+   Deep Bayesian Active Learning for Accelerating Stochastic Simulation.
+   arXiv preprint arXiv:2106.02770. Retrieved from https://arxiv.org/abs/2106.02770
+
+Contributor: eibarolle
+"""
+
+from __future__ import annotations
+
+import warnings
+from typing import Optional
+
+import torch
+from botorch.acquisition import AcquisitionFunction
+from botorch_community.models.np_regression import NeuralProcessModel
+from torch import Tensor
+
+import torch
+#reference: https://arxiv.org/abs/2106.02770 
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+class LatentInformationGain(AcquisitionFunction):
+    def __init__(
+        self, 
+        context_x: torch.Tensor, 
+        context_y: torch.Tensor,
+        model: NeuralProcessModel, 
+        num_samples: int = 10,
+        min_std: float = 0.01,
+        scaler: float = 0.5
+    ) -> None:
+        """
+        Latent Information Gain (LIG) Acquisition Function, designed for the
+        NeuralProcessModel. This is a subclass of AcquisitionFunction.
+
+        Args:
+            model: Trained NeuralProcessModel.
+            context_x: Context input points, as a Tensor.
+            context_y: Context target points, as a Tensor.
+            num_samples (int): Number of samples for calculation, defaults to 10.
+            min_std: Float representing the minimum possible standardized std, defaults to 0.1.
+            scaler: Float scaling the std, defaults to 0.9.
+        """
+        super().__init__(model=model)
+        self.model = model.to(device)
+        self.num_samples = num_samples
+        self.min_std = min_std
+        self.scaler = scaler
+        self.context_x = context_x.to(device)
+        self.context_y = context_y.to(device)
+
+    def forward(self, candidate_x):

```

Comment:
It seems to me like the acquisition function computes the information gain on one batch of points, with the batch being in the first dimension. Thus, the output of this forward would be one scalar. 

This would run contrary to the acquisition function convention, and so it wouldn't be able to be used with `optimize_acqf` etc.

Reply:
You would want the forward to be able to handled a `N x q x D`-shaped input, where you are currently only computing the `q`-element (but that aspect seems correct as far as I can tell right now). This may be a bit challenging, but certainly looks doable!